src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1409,37 +1452,33 @@ void Ekf2::run()
 					lpos.hagl_max = INFINITY;
 				}
 
-				// Get covariances to vehicle odometry
-				float covariances[24];
-				_ekf.get_covariances(covariances);
+				// get pose covariance and copy the URT to the uORB message
+				float *pos_p = &odom.pose_covariance[0];
 
-				// get the covariance matrix size
-				const size_t POS_URT_SIZE = sizeof(odom.pose_covariance) / sizeof(odom.pose_covariance[0]);
-				const size_t VEL_URT_SIZE = sizeof(odom.velocity_covariance) / sizeof(odom.velocity_covariance[0]);
+				// get the position and orientation covariances and fil the pose covariance matrix
+				// it's not a cross-covariance matrix but simplifies propagating the data
+				matrix::SquareMatrix<float, 6> pose_cov = matrix::eye<float, 6>() * 9999.0f;
```

Comment:
Why `9999.0f`?

Reply:
You don't estimate angular velocities. So you should pass an high value for the covariance since it's unknown.

Reply:
I know, I'm questioning the user of a magic number. Why not set it to `INFINITY` or `NAN` to convey you don't know?

Reply:
The problem is how it has to be interpreted on the Mavlink side. One thing is the covariance being unknown, the other is the estimate being unknown. It should be uniform as well between Mavlink and the ROS definitions.

Reply:
I am sending `NAN` for now. But I will have to review the conventions here. Before I considered that only the first element of the matrix was to be set to `NAN`. But I think it makes more sense to, by default, set the diagonal to `NAN` and then handle these accordingly. I will do a review of this throughout the code on a further PR.